jujugui/static/gui/src/app/utils/environment-change-set.js

```diff
@@ -711,7 +711,8 @@ YUI.add('environment-change-set', function(Y) {
       var db = this.get('db');
       var machine = db.machines.getById(command.args[0]);
       var removedUnits = [];
-      machine.units.forEach(function(unit) {
+      var units = db.units.filterByMachine(machine.id);
+      units.forEach(function(unit) {
```

Comment:
As above, removed this kinda hacky unit list which relied on the machine view having been opened before a machine was destroyed, we can just do a lookup using the machine id.